pkg/services/store/http.go

```diff
@@ -25,35 +27,47 @@ func ProvideHTTPService(store StorageService) HTTPStorageService {
 }
 
 func (s *httpStorage) Upload(c *models.ReqContext) response.Response {
-	action := "Upload"
-	scope, path := getPathAndScope(c)
+	// 32 MB is the default used by FormFile()
+	if err := c.Req.ParseMultipartForm(32 << 20); err != nil {
```

Comment:
We are not limiting upload size here - it's only a memory limit, the rest of the body will be put on disk right?  So maybe we need https://pkg.go.dev/net/http#MaxBytesReader or sth. Or maybe it's already set somewhere in Grafana? Also I don't understand the correlation between this limit and max upload size?

Just feel anxious about this - maybe it's already fine.

Reply:
No we don't limit the upload size here, but I tried here:
https://github.com/grafana/grafana/pull/46749/files#diff-f769842bf966d8b9a4e63574f38ce36fdbbebc534d6579536f2445b8f4a1b1baR137-R142

Reply:
I think header information can be spoofed